packages/visual-editor/src/components/Dropzone/EditorBlock.tsx

```diff
@@ -59,21 +60,25 @@ export const EditorBlock: React.FC<EditorBlockProps> = ({
   const containsZone = sectionsWithZone[componentId];
 
   const isAssemblyBlock = node.type === ASSEMBLY_BLOCK_NODE_TYPE;
+  const isAssembly = node.type === ASSEMBLY_NODE_TYPE;
 
   const onClick = (e: React.SyntheticEvent<Element, Event>) => {
     e.stopPropagation();
 
-    if (isAssemblyBlock && !containsZone) {
-      // Readonly components in an assembly cannot be selected
-      return;
-    }
-    const nodeId = isAssemblyBlock ? parentSectionId : componentId;
-
-    // Only select the node if the user intentionally clicked on it, but not when dragging
     if (!userIsDragging) {
-      setSelectedNodeId(nodeId);
+      // if it is the assembly directly we just want to select it as a normal component
+      if (isAssembly) {
+        setSelectedNodeId(node.data.id);
+        sendMessage(OUTGOING_EVENTS.ComponentSelected, {
+          nodeId: node.data.id,
+        });
+        return;
+      }
+
+      setSelectedNodeId(node.data.id);
       sendMessage(OUTGOING_EVENTS.ComponentSelected, {
-        nodeId,
+        assembly: node.data.assembly,
```

Comment:
it looks like we now send this not only for assembly blocks but also normal nodes. Is this intended as `assembly` will be undefined in the latter scenario? If yes, then we can remove the unused variable `isAssemblyBlock` if I'm not mistaken.

Reply:
in line 71 we send the event without the assembly, so there is a difference between when it is an assembly block or not. Or am missunderstanding it? 🤔

Reply:
In line 71, we send the event when the node itself it an `assembly`, here it is either an `assembly block` or just a normal node. I was wondering about this third case.